### PR TITLE
[POC - Behat] Add RepositoryContext trait

### PR DESCRIPTION
**Pull request by miguelcleverti**
As decided BehatBundle is to be deprecated and separated between the kernel and platformUI bundle. The idea is to make the Behat easier to maintain, having less repo dependencies should help releases and versions also.

This PR moves the RepositoryContext trait into the kernel. This is trait contains all the methods related to the repo . Since this is to be used by any contexts that need the repo it should be in the kernel to be easily included and not duplicate.

Related PRs:
* Behat browser context #1585
* Add EzBehatExtension and ArgumentResolver #1586
* Deprecate Behat classes https://github.com/ezsystems/BehatBundle/pull/48

### CODE REVIEW

**Comment by miguelcleverti** [comment]
ping @bdunogier @andrerom @joaoinacio

File: eZ/Bundle/PlatformBehatBundle/Context/RepositoryContext.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+trait RepositoryContext
+{
+    /**
+     * Default Administrator user id.
+     */
+    private $adminUserId = 14;
+
+    /**
+     * @var Repository
```
Comment (bdunogier) [inline eZ/Bundle/PlatformBehatBundle/Context/RepositoryContext.php]: Please use the full namespace in phpdoc (same below).

File: eZ/Bundle/PlatformBehatBundle/Context/RepositoryContext.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+namespace EzSystems\PlatformBehatBundle\Context;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\Core\Repository\Values\User\UserReference;
+
+/**
+ * Repository Context Trait.
```
Comment (bdunogier) [inline eZ/Bundle/PlatformBehatBundle/Context/RepositoryContext.php]: Please elaborate a bit about how it is used. I'd like to see at least one reference to a place where it's used, and a simplified example.

**Comment by miguelcleverti** [comment]
@bdunogier Improved the docs in the Context and added a more detailed to a dedicated behat docs file.

**Comment by bdunogier** [comment]
How is the repository injected in the trait again ?

**Comment by miguelcleverti** [comment]
@bdunogier the repository has to be injected into the context in which this trait is used, using the ArgumentResolver (#1586), pass it through the behat.yml into the constructor or any other way...

File: eZ/Bundle/PlatformBehatBundle/Context/RepositoryContext.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    /**
+     * @var eZ\Publish\API\Repository\Repository
+     */
+    protected $repository;
+
+    /**
+     * @param eZ\Publish\API\Repository\Repository $repository
+     */
+    public function setRepository(Repository $repository)
```
Comment (bdunogier) [inline eZ/Bundle/PlatformBehatBundle/Context/RepositoryContext.php]: This should be protected, not public. Each context that uses the trait would expose setRepository & loginAdmin, while it is not necessary.

**Comment by bdunogier** [comment]
I'm not sure what to say. It is fine, and it works, but for many little reaons it does not sound right...

But let's say that if you:
- fix the visibility
- add a getter for `$repository` and make the property private

I'll be +1 then.

**Comment by miguelcleverti** [comment]
@dpobel changed the visibility and added the getter

**Comment by miguelcleverti** [comment]
Actually the `loginAdmin()` must be `public` or else behat can't use it as for the `BeforeScenario` Hook

**Comment by miguelcleverti** [comment]
ping @bdunogier

**Comment by bdunogier** [comment]
And it is merged. Let's see during the 16.06 cycle how it behaves. I'll use it in a scenario of mine right away.

**Pull request merged**